app/Models/PrintJob.php

```diff
@@ -37,46 +42,92 @@ class PrintJob extends Model
 {
     use HasFactory;
 
-    protected $table = 'print_jobs';
-    protected $primaryKey = 'id';
-    public $incrementing = true;
     public $timestamps = true;
 
-    public const QUEUED = 'QUEUED';
-    public const ERROR = 'ERROR';
-    public const CANCELLED = 'CANCELLED';
-    public const SUCCESS = 'SUCCESS';
-    public const STATES = [
-        self::QUEUED,
-        self::ERROR,
-        self::CANCELLED,
-        self::SUCCESS,
+    protected $fillable = [
+        'user_id',
+        'state',
+        'job_id',
+        'cost',
+        'printer_id',
+        'used_free_pages',
+        'filename',
     ];
 
-    protected $fillable = [
-        'filename', 'filepath', 'user_id', 'state', 'job_id', 'cost',
+    protected $casts = [
+        'state' => PrintJobStatus::class,
+        'used_free_pages' => 'boolean',
     ];
 
     public function user()
     {
-        return $this->belongsTo('App\Models\User');
+        return $this->belongsTo(User::class);
+    }
+
+    /**
+     * `Printer` which this `PrintJob` was sent to.
+     * @return BelongsTo
+     */
+    public function printer()
+    {
+        return $this->belongsTo(Printer::class);
     }
 
-    public static function translateStates(): \Closure
+    /**
+     * `PrintAccount` which is related to this `PrintJob` through the `User`.
+     * The `PrintJob` and the `PrintAccount` both belong to the `User`, in this sense this relationship is articifial.
+     * Trying to fix the decision made for the database a few years ago.
+     * @return HasOneThrough
+     */
+    public function printAccount()
     {
-        return function ($data) {
-            $data->state = __('print.'.strtoupper($data->state));
+        return $this->hasOneThrough(
+            PrintAccount::class,
+            User::class,
+            'id', // Foreign key on users
+            'user_id', // Foreign key on print_accounts
+            'user_id', // Local key on print_jobs
+            'id', // Local key on users
+        );
+    }
 
-            return $data;
-        };
+    /**
+     * Attribute for the translated cost.
+     */
+    public function getTranslatedCostAttribute()
+    {
+        return $this->used_free_pages ? "$this->cost ingyenes oldal" : "$this->cost HUF";
+    }
+
+    /**
+     * Attribute for the translated state.
+     */
+    public function getTranslatedStateAttribute()
+    {
+        return __("print." . strtoupper($this->state->value));
     }
 
-    public static function addCurrencyTag(): \Closure
+    /**
+     * Attemts to cancel the given `PrintJob`. Returns wether it was successful.
+     * @param PrintJob $this
+     * @return PrinterCancelResult
+     */
+    public function cancel()
     {
-        return function ($data) {
-            $data->cost = "{$data->cost} HUF";
+        $printer = $this->printer ?? Printer::firstWhere('name', config('print.printer_name'));
```

Comment:
Again, why don't we always have a printer set in the db? I would delete the config entry or use only that to avoid confusion.